tests/JavaTest.java

```diff
@@ -121,6 +135,17 @@ public static void main(String[] args) {
         // the mana field should retain its default value
         TestEq(monster.mutateMana((short)10), false);
         TestEq(monster.mana(), (short)150);
+		
+		// Accessing a vector of sorted by the key tables
+        Monster[] monsters = { monster.testarrayoftables(0), monster.testarrayoftables(1), monster.testarrayoftables(2) };
```

Comment:
So `lookupByKey` should get a vector offset as an argument?
`public static Monster lookupByKey(int vectorOffset, String key)`